src/iso.rs

```diff
@@ -665,140 +675,146 @@ impl IsoTime {
     /// Rounds the current `IsoTime` according to the provided settings.
     pub(crate) fn round(
         &self,
-        increment: RoundingIncrement,
-        unit: TemporalUnit,
-        mode: TemporalRoundingMode,
-        day_length_ns: Option<u64>,
+        resolved_options: ResolvedRoundingOptions,
     ) -> TemporalResult<(i32, Self)> {
-        // 1. Let fractionalSecond be nanosecond × 10-9 + microsecond × 10-6 + millisecond × 10-3 + second.
-
-        let quantity = match unit {
-            // 2. If unit is "day", then
-            // a. If dayLengthNs is not present, set dayLengthNs to nsPerDay.
-            // b. Let quantity be (((((hour × 60 + minute) × 60 + second) × 1000 + millisecond) × 1000 + microsecond) × 1000 + nanosecond) / dayLengthNs.
-            // 3. Else if unit is "hour", then
-            // a. Let quantity be (fractionalSecond / 60 + minute) / 60 + hour.
-            TemporalUnit::Hour | TemporalUnit::Day => {
-                u64::from(self.nanosecond)
-                    + u64::from(self.microsecond) * 1_000
-                    + u64::from(self.millisecond) * 1_000_000
-                    + u64::from(self.second) * 1_000_000_000
-                    + u64::from(self.minute) * 60 * 1_000_000_000
-                    + u64::from(self.hour) * 60 * 60 * 1_000_000_000
+        // 1. If unit is "day" or "hour", then
+        let quantity = match resolved_options.smallest_unit {
+            TemporalUnit::Day | TemporalUnit::Hour => {
+                // a. Let quantity be ((((hour × 60 + minute) × 60 + second) × 1000 + millisecond)
+                // × 1000 + microsecond) × 1000 + nanosecond.
+                ((((i128::from(self.hour) * 60 + i128::from(self.minute)) * 60
+                    + i128::from(self.second))
+                    * 1000
+                    + i128::from(self.millisecond))
+                    * 1000
+                    + i128::from(self.microsecond))
+                    * 1000
+                    + i128::from(self.nanosecond)
```

Comment:
I think this could benefit from separating it on several variables, even if they're just used once. Also applies to the other branches that do a big operation like this.

Reply:
Yeah, it would probably help with the readability.